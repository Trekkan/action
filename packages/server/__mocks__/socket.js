/* eslint-env jest */
const socket = {
  emit: jest.fn(),
  setAuthToken: jest.fn()
}

export default socket
