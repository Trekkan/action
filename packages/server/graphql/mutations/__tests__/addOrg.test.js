/* eslint-env jest */
import DynamicSerializer from 'dynamic-serializer'
import MockDate from 'mockdate'
import socket from '../../../__mocks__/socket'
import makeDataLoader from '../../../__tests__/setup/makeDataLoader'
import mockAuthToken from '../../../__tests__/setup/mockAuthToken'
import MockDB from '../../../__tests__/setup/MockDB'
import {__now} from '../../../__tests__/setup/mockTimes'
import fetchAndSerialize from '../../../__tests__/utils/fetchAndSerialize'
import getRethink from '../../../database/rethinkDriver'
import addOrg from '../addOrg'

MockDate.set(__now)
console.error = jest.fn()

describe('addOrg', () => {
  test('adds a new org with no invitees', async () => {
    // SETUP
    const r = await getRethink()
    const dynamicSerializer = new DynamicSerializer()
    const mockDB = new MockDB()
    const {organization, user} = await mockDB.init().organization(0)
    const org = organization[0]
    const authToken = mockAuthToken(user[0])
    const dataLoader = makeDataLoader(authToken)

    // TEST
    const newTeam = {
      name: 'addOrg|1|NewTeamName'
    }
    const orgName = 'addOrg|1|NewOrgName'
    const res = await addOrg.resolve(undefined, {newTeam, orgName}, {authToken, dataLoader, socket})

    const {orgId, teamId} = res
    // VERIFY
    const db = await fetchAndSerialize(
      {
        organization: r
          .table('Organization')
          .getAll(orgId, {index: 'id'})
          .orderBy('name'),
        team: r
          .table('Team')
          .getAll(orgId, {index: 'orgId'})
          .orderBy('name'),
        teamMember: r
          .table('TeamMember')
          .getAll(teamId, {index: 'teamId'})
          .orderBy('preferredName'),
        user: r
          .table('User')
          .getAll(org.id, orgId, {index: 'userOrgs'})
          .orderBy('preferredName')
      },
      dynamicSerializer
    )
    expect(db).toMatchSnapshot()
  })
})
