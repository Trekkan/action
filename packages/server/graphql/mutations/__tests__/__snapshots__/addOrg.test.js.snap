// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addOrg adds a new org with no invitees 1`] = `
Object {
  "organization": Array [
    Object {
      "createdAt": 2017-01-01T08:00:00.000Z,
      "creditCard": Object {},
      "id": 0,
      "name": "addOrg|1|NewOrgName",
      "orgUsers": Array [
        Object {
          "id": 1,
          "inactive": false,
          "role": "billingLeader",
        },
      ],
      "tier": "personal",
      "updatedAt": 2017-01-01T08:00:00.000Z,
    },
  ],
  "team": Array [
    Object {
      "activeFacilitator": null,
      "facilitatorPhase": "lobby",
      "facilitatorPhaseItem": null,
      "id": 2,
      "isArchived": false,
      "isPaid": true,
      "meetingId": null,
      "meetingPhase": "lobby",
      "meetingPhaseItem": null,
      "name": "addOrg|1|NewTeamName",
      "orgId": 0,
      "tier": "personal",
    },
  ],
  "teamMember": Array [
    Object {
      "checkInOrder": 4,
      "email": "steve.rogers@avengers.org",
      "id": 3,
      "isCheckedIn": null,
      "isFacilitator": true,
      "isLead": true,
      "isNotRemoved": true,
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Captain America",
      "teamId": 2,
      "userId": 1,
    },
  ],
  "user": Array [
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "natasha.romanova@avengers.org",
      "emailVerified": false,
      "id": 5,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Natasha Romanova",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Black Widow",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "steve.rogers@avengers.org",
      "emailVerified": false,
      "id": 1,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Steve Rogers",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Captain America",
      "tms": Array [
        12,
        2,
      ],
      "updatedAt": 2017-01-01T08:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": "billingLeader",
        },
        Object {
          "id": 14,
          "role": null,
        },
        Object {
          "id": 0,
          "role": "billingLeader",
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "steve.rogers@avengers.org",
      "emailVerified": false,
      "id": 1,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Steve Rogers",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Captain America",
      "tms": Array [
        12,
        2,
      ],
      "updatedAt": 2017-01-01T08:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": "billingLeader",
        },
        Object {
          "id": 14,
          "role": null,
        },
        Object {
          "id": 0,
          "role": "billingLeader",
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "bruce.banner@avengers.org",
      "emailVerified": false,
      "id": 6,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Bruce Banner",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Hulk",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "tony.stark@avengers.org",
      "emailVerified": false,
      "id": 7,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Tony Stark",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Iron Man",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "kyle.richmond@avengers.org",
      "emailVerified": false,
      "id": 8,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Kyle Richmond",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Nighthawk",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "thor@avengers.org",
      "emailVerified": false,
      "id": 9,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Thor Odinson",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Thor",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": "billingLeader",
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "vision@avengers.org",
      "emailVerified": false,
      "id": 10,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "Jonas",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "Vision",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
    Object {
      "cachedAt": 2017-01-01T07:00:00.000Z,
      "createdAt": 2017-01-01T07:00:00.000Z,
      "email": "james.rhodes@avengers.org",
      "emailVerified": false,
      "id": 11,
      "identities": Array [],
      "inactive": false,
      "lastLogin": 2017-01-01T07:00:00.000Z,
      "lastSeenAt": 2017-01-01T07:00:00.000Z,
      "name": "James Rhodes",
      "picture": "https://placeimg.com/100/100/animals",
      "preferredName": "War Machine",
      "tms": Array [
        12,
      ],
      "updatedAt": 2017-01-01T07:00:00.000Z,
      "userOrgs": Array [
        Object {
          "id": 13,
          "role": null,
        },
        Object {
          "id": 14,
          "role": null,
        },
      ],
      "welcomeSentAt": 2017-01-01T07:00:00.000Z,
    },
  ],
}
`;
